ui/src/common/containers/EditRecordActionContainer.jsx

```diff
@@ -1,20 +1,15 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-import { Set } from 'immutable';
 
-import { isCataloger } from '../authorization';
 import ListItemAction from '../components/ListItemAction';
 import IconText from '../components/IconText';
 import EventTracker from '../components/EventTracker';
 import ExternalLink from '../components/ExternalLink';
 
 class EditRecordActionContainer extends Component {
   render() {
-    const { userRoles, recordId } = this.props;
-    const isUserCataloger = isCataloger(userRoles);
-    const href = `/workflows/edit_article/${recordId}`;
-    if (isUserCataloger) {
+    const { href, canDisplay } = this.props;
+    if (canDisplay && href) {
```

Comment:
It is unusual to put to logic as visibility logic that directly (It is OK in cases like, `InlineList` component doesn't render when `items` is an empty array) here in the component. You could put this in the parent {canEdit && <EditRecordAction ... />}`

Reply:
On the other hand, since this is the `Edit` button. We can pass `pidType`, `pidValue` and push the logic of generating url for editing in this component instead of doing it at the parent level. For example have a map of `{ pidType: editUrlPrefix }` and ```href={`${pidTypeToEditUrlPrefix[pidType]}/${pidValue}`}```